katello/katello/hostname-change.rb

```diff
@@ -369,18 +381,30 @@ def run
       installer << " --disable-system-checks" if disable_system_check_option?
 
       STDOUT.puts installer
-      installer_output = self.run_cmd("#{installer}")
-      installer_success = $?.success?
-      STDOUT.puts installer_output
-
-      STDOUT.puts "Restarting puppet services"
-      self.run_cmd("/sbin/service puppet restart")
-      self.run_cmd("katello-service restart --only puppetserver")
+      default_installer << ' --disable-system-checks' if disable_system_check_option?
+      failure_message = """
+Something went wrong with the #{@options[:scenario].capitalize} installer.
+Please check the above output and the corresponding logs. Once the issue is resolved
+you may complete the hostname change by running #{default_installer}
+      """
+
+      run_cmd(installer, [0], failure_message) do |result, success|
+        if temp_last_scenario_yaml && temp_last_scenario_yaml.path
```

Comment:
Wouldn't this variable be undefined if the file didn't exist?

Reply:
Correct. When a tempfile is issued `.unlink` all traces are removed and it will return nil which is what I am checking for here. If it responds to path (which we need) then we know it hasn't been unlinked.